presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -341,10 +353,20 @@ public Page getNextPage()
         readPositions += batchSize;
 
         initializePositions(batchSize);
+        initalizeErrors(batchSize);
 
         int[] positionsToRead = this.positions;
         int positionCount = batchSize;
-        boolean filterFunctionsApplied = filterFunctions.isEmpty();
+
+        if (filterFunctionWithoutInput.isPresent()) {
+            positionCount = applyFilterFunctionsWithNoInputs(positionsToRead, positionCount);
+        }
+        boolean filterFunctionsApplied = filterFunctionsWithInputs.isEmpty();
+
+        if (positionCount == 0) {
```

Comment:
- move this inside the `if (filterFunctionWithoutInput.isPresent())` block
- call `batchRead(batchSize);` before returning